.github/actions/setup-eks-cluster/action.yml

```diff
@@ -71,13 +71,18 @@ runs:
 
             echo "Trying kubeconfig: ${KUBECONFIG_KEY}"
 
-            # Try to download and delete atomically
-            KUBECONFIG_PATH="/tmp/kubeconfig-${RANDOM}"
-
-            # Download the kubeconfig
-            if aws s3 cp "s3://${S3_BUCKET}/${S3_PREFIX}${KUBECONFIG_KEY}" "${KUBECONFIG_PATH}" --region ${{ inputs.region }}; then
-              # Delete it from S3 to prevent others from using it
-              if aws s3 rm "s3://${S3_BUCKET}/${S3_PREFIX}${KUBECONFIG_KEY}" --region ${{ inputs.region }}; then
+            # Generate a unique claim ID for this job run
+            CLAIM_ID="${{ github.run_id }}-${{ github.run_attempt }}-${RANDOM}"
```

Comment:
NIT: `${{ github.run_id }}-${{ github.run_attempt }}` should be enough, isn't it?